core/src/main/scala/org/apache/spark/TaskContext.scala

```diff
@@ -184,4 +184,13 @@ abstract class TaskContext extends Serializable {
    */
   private[spark] def registerAccumulator(a: Accumulable[_, _]): Unit
 
+  /**
+   * Set the current RDD and partition being processed.
+   */
+  private[spark] def setRDDPartitionInfo(rddId: Int, index: Int): Unit
+
+  /**
+   * Returns the current RDD and Partition ids being processed. May be null.
+   */
+  private[spark] def getRDDPartitionInfo(): (Int, Int)
```

Comment:
why would this be null?

I'd also add a comment that this can be used to track whether an rdd has been computed previously -- if it gets recomputed for any reason, the ids will be the same.  (there are so many different types of ids running around I think its easy to get confused.)